02_knight_and_princess/gleb_skryabin/graph.hpp

```diff
@@ -0,0 +1,101 @@
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <set>
+#include <vector>
+
+class Graph {
+ public:
+  struct Edge {
+    int id;
+    std::pair<int, int> vertices;
+  };
+
+  struct Vertex {
+    int id;
+    std::set<int> edges;
+  };
+
+  Graph(std::vector<Edge> inpEdges) {
+    for (auto edge = inpEdges.begin(); edge != inpEdges.end(); edge++) {
+      auto e = edges.find(edge->id);
+      if (e == edges.end()) {
+        // add edge
+        edges.emplace(edge->id, *edge);
+      } else {
+        // edit edge
+        e->second.vertices = edge->vertices;
+      }
+    }
+    for (auto edge = edges.begin(); edge != edges.end(); edge++) {
+      auto vs = edge->second.vertices;
+      int vsid[] = {vs.first, vs.second};
+      for (int i = 0; i < 2; i++) {
+        auto v = vertices.find(vsid[i]);
+        if (v == vertices.end()) {
+          // add vertex
+          Vertex newVertex = {vsid[i], {edge->first}};
```

Comment:
Лучше использовать конструктор с именованными аргументами:
```suggestion
          Vertex newVertex = { .id = vsid[i], .edges = {edge->first}};
```